pkg/cmd/recoveryapiserver/create.go

```diff
@@ -0,0 +1,127 @@
+package recoveryapiserver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/client-go/tools/watch"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/recovery"
+)
+
+type CreateOptions struct {
+	Options
+
+	KubeApiserverImage    string
+	StaticPodResourcesDir string
+	Timeout               time.Duration
+	Wait                  bool
+}
+
+func NewCreateCommand() *cobra.Command {
+	o := &CreateOptions{
+		Options:               NewDefaultOptions(),
+		KubeApiserverImage:    "", // TODO: set the public pullspec
+		StaticPodResourcesDir: "/etc/kubernetes/static-pod-resources",
+		Timeout:               5 * time.Minute,
+		Wait:                  true,
+	}
+
+	cmd := &cobra.Command{
+		Use: "create",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Complete()
+			if err != nil {
+				return err
+			}
+
+			err = o.Validate()
+			if err != nil {
+				return err
+			}
+
+			err = o.Run()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	// TODO: add/prettify descriptions
+	AddFlags(cmd, &o.Options)
+
+	cmd.Flags().StringVar(&o.KubeApiserverImage, "kube-apiserver-image", o.KubeApiserverImage, "image reference for kubernetes apiserver")
+	cmd.Flags().StringVar(&o.StaticPodResourcesDir, "static-pod-resources", o.StaticPodResourcesDir, "path to store a directory containing static pod resources for recovery apiserver manifest")
+	cmd.Flags().DurationVar(&o.Timeout, "timeout", o.Timeout, "startup timeout, 0 means infinite")
+	cmd.Flags().BoolVar(&o.Wait, "wait", o.Wait, "wait for recovery apiserver to become ready")
+
+	return cmd
+}
+
+func (o *CreateOptions) Complete() error {
+	err := o.Options.Complete()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *CreateOptions) Validate() error {
+	err := o.Options.Validate()
+	if err != nil {
+		return err
+	}
+
+	if o.KubeApiserverImage == "" {
+		return errors.New("missing required flag --kube-apiserver-image")
+	}
+
+	if o.StaticPodResourcesDir == "" {
+		return errors.New("missing required flag --static-pod-resources")
+	}
+
+	return nil
+}
+
+func (o *CreateOptions) Run() error {
+	ctx, cancel := watch.ContextWithOptionalTimeout(context.Background(), o.Timeout)
```

Comment:
context.TODO()

Reply:
in future we will pass the context via Run() that context will be wired to TERM signal handler